telosnetwork/teloscan
# Improve Address Page Display and Contract Manager Optimization

Fixes #829 #860

## Description
- Added blockie icons to address/contract/token pages for better visual identification
- Replaced generic icons with blockie-icons for contracts and normal addresses
- Retained token icons for ERC-20 tokens
- Increased field width for function names to show complete names
- Optimized contract manager by removing null address list cache
- Improved performance by using cached contract metadata from indexer API queries
- Eliminated redundant API calls to `/v1/contract/{address}`

## Test scenarios
1. Address Icons:
   - Verify blockie icon display at: `/address/0x77F0Db20D64051D081170e95fC0d0F5aa40C249c`
   - Check contract icon at: `/address/0x3B7023f3C99EAe2FA0AaeB9490A78d5D98a24148`
   - Confirm token icon at: `/address/0xB4B01216a5Bc8F1C8A33CD990A1239030E60C905?tab=transactions`

2. Contract Manager:
   - Clear site data in Dev Tools (Applications Tab > Storage)
   - Navigate to address 0x0e79065B5F11b5BD1e62B935A600976ffF3754B9
   - Verify no additional calls to `/v1/contract/{address}`
   - Check Transactions Page for absence of redundant contract metadata queries

## Checklist:
- [x] I have commented code for complex icon generation logic
- [x] I have tested for mobile functionality
- [x] Performance improvements have been verified
- [x] UI changes have been tested across different address types
- [x] Cache optimization has been thoroughly tested
